src/main/scala/controllers/ActionHandler.scala

```diff
@@ -102,27 +102,22 @@ class ActionHandler @Inject()(applications: ApplicationOps, applicationForms: Ap
   def redirectToPreview(id: ApplicationId, sectionNumber: Int) =
     Redirect(routes.ApplicationPreviewController.previewSection(id, sectionNumber))
 
-  def renderSectionForm(id: ApplicationId,
+  def renderSectionForm(app: ApplicationDetail,
```

Comment:
I'm not clear what you mean by "sync context". It's just a pure function that generates a Result.